examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py

```diff
@@ -36,8 +35,7 @@ def build_airflow_polling_sensor(
         name="airflow_dag_status_sensor",
         minimum_interval_seconds=1,
         default_status=DefaultSensorStatus.RUNNING,
-        # Tracking bug around this here: https://linear.app/dagster-labs/issue/FOU-376/make-airflow-polling-sensor-work-with-asset-selection=
```

Comment:
Ah good call. It's actually more correct anyway to target an empty selection since IIUC this sensor never wants to trigger any materializations.

Updated with a comment.

Reply:
The meaning of asset_selection is really quite strange. Asset selections can target checks, but the selection is the empty set, but the run request can target checks?

Reply:
Yeah this is a function of checks being "not fully implemented" wrt asset checks. The check in `_run_requests_with_base_asset_jobs` (the function modified in this PR) that checks that targeted asset keys are part of the sensor `asset_selection` isn't run for asset checks.

I think the right solution here is to:

- modify that check so that it _is_ run for asset checks.
- Pass the selected checks instead of run request asset selection instead of the empty set.

I'll make those changes.

Reply:
OK so I have ended up changing this to a form that (a) makes more sense; (b) will work with older versions of Dagster.

What made `asset_selection="*"` fail on older dagster versions is that the `asset_selection=[]` was being ignored when generating the run request, resulting in a request that targeted _all_ assets-- including the unmaterializable airflow assets (selected by `*`).

I've now made two changes:

- the sensor targets `AssetSelection.all_asset_checks()`, so the sensor can never launch runs trying to materialize unmaterializable assets
- I removed the explicitly passed `asset_selection=[]` on the `RunRequest`. That's fine, the sensor will fill it in with the sensor's asset selection, which is empty because we now target only `all_asset_checks()`.

Reply:
So this will still work on older versions of Dagster?

Reply:
>So this will still work on older versions of Dagster?

Yes, it should

Reply:
Now that we have the warning I'm more comfortable with this, thanks.